### PR TITLE
[SPARK-33364][SQL] Introduce the "purge" option in TableCatalog.dropTable for v2 catalog.

### PR DESCRIPTION
**Pull request by imback82**
### What changes were proposed in this pull request?

This PR proposes to introduce the `purge` option in `TableCatalog.dropTable` so that v2 catalogs can use the option if needed.

Related discussion: https://github.com/apache/spark/pull/30079#discussion_r510594110

### Why are the changes needed?

Spark DDL supports passing the purge option to `DROP TABLE` command. However, the option is not used (ignored) for v2 catalogs.

### Does this PR introduce _any_ user-facing change?

This PR introduces a new API in `TableCatalog`.

### How was this patch tested?

Added a test.

### CODE REVIEW

**Comment by imback82** [comment]
cc @rdblue @cloud-fan

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @imback82 .

File: sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/TableCatalog.java [hunk 1]
```diff
@@ -162,6 +162,29 @@ Table alterTable(
...
 
+  /**
+   * Drop a table in the catalog with an option to purge.
+   * <p>
+   * If the catalog supports views and contains a view for the identifier and not a table, this
+   * must not drop the view and must return false.
+   * <p>
+   * If the purge option is set to false, the default implementation falls back to
+   * {@link #dropTable(Identifier)} dropTable}. Otherwise, it throws
+   * {@link UnsupportedOperationException}.
```
Comment (cloud-fan) [inline sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/TableCatalog.java:173]: we should mention that, people need to overwrite this method to support the purge option.

**Review by cloud-fan — APPROVED** [review]

File: sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/TableCatalog.java [hunk 2]
```diff
@@ -162,6 +162,29 @@ Table alterTable(
    */
   boolean dropTable(Identifier ident);
 
+  /**
+   * Drop a table in the catalog with an option to purge.
+   * <p>
+   * If the catalog supports views and contains a view for the identifier and not a table, this
+   * must not drop the view and must return false.
+   * <p>
+   * If the catalog supports the option to purge a table, this method must be overwritten.
```
Comment (dongjoon-hyun) [inline sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/TableCatalog.java:171]: `overwritten` -> `overridden`.

**Review by dongjoon-hyun — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Merged to master. The last two commits are only for comment changes.

**Comment by imback82** [comment]
Thanks @dongjoon-hyun / @cloud-fan for the review!

**Comment by rdblue** [comment]
Late +1 from me. Thanks @imback82!

**Pull request closed without merging**